### ISSUE TITLE
Typescript error on v9

### ISSUE THREAD

**Issue by eugenenelou** [jacobwgillespie/styled-icons#1077]
Hello, I have a typescript error when I try upgrading to v9.
```
TypeScript error in /front/node_modules/styled-icons/types/types.ts(2,9):
Cannot re-export a type when the '--isolatedModules' flag is provided.  TS1205
   1 | import {StyledIcon, StyledIconProps} from '@styled-icons/styled-icon'
> 2 | export {StyledIcon, StyledIconProps}
```

`--isolatedModules` is required for babel compilation of typescript.

**Comment by jacobwgillespie** [comment]
Thanks for the report, one question before I look into a solution, are you directly importing `styled-icons/types` in your codebase?  Or does this error happen with just an icon import?

**Comment by eugenenelou** [comment]
Yes I'm trying to type a props as a `StyledIcon`

**Comment by JClackett** [comment]
I have a similar issue with v9 and TS

Property 'crossOrigin' is missing in type '{ size: number; }' but required in type 'Pick<StyledIconProps

**Comment by jacobwgillespie** [comment]
@eugenenelou and @JClackett can I get a bit more information on which version of TypeScript and `@types` packages you are using?  I'm not able to directly reproduce this issue.  Or even better, if you have a repo or CodeSandbox that reproduces the issue, that would be amazing.

@eugenenelou, the Styled Icons website is built with Next.js using Babel, with `--isolatedModules` enabled, and it imports from `styled-icons/types` without issue.  However, I'm not sure _why_ it doesn't have an issue, so still investigating.

@JClackett, that sounds like a mismatch between `@types/react`, `@types/styled-components` and `styled-icons`.  I can better investigate if you have more info on how to reproduce (a repo, dependency list, etc).

**Comment by gilesbradshaw** [comment]
I wonder if this will help..  

https://github.com/gilesbradshaw/ui-example

If I comment out import App from './App it says:

```
Failed to compile.

/home/giles/ui-example/ui/src/App/Created/index.tsx
TypeScript error in /home/giles/ui-example/ui/src/App/Created/index.tsx(143,15):
No overload matches this call.
  Overload 1 of 2, '(props: Pick<Pick<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "type" | "name" | "className" | "color" | "height" | "id" | "lang" | ... 456 more ... | "size"> & RefAttributes<...>, "string" | ... 469 more ... | "size"> & Partial<...>, "string" | ... 469 more ... | "size"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Type 'true' is not assignable to type '"" | "anonymous" | "use-credentials" | undefined'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<ForwardRefExoticComponent<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "type" | "name" | "className" | "color" | "height" | "id" | "lang" | ... 456 more ... | "size"> & RefAttributes<...>>, any, {}, never>): ReactElement<...>', gave the following error.
    Type 'true' is not assignable to type '"" | "anonymous" | "use-credentials" | undefined'.  TS2769

    141 |           >
    142 |             <Up
  > 143 |               crossOrigin={true}
        |               ^
    144 |             />
    145 |           </Link>
    146 |           : <Link

```
if I don;t comment out import App from './App'

it says:

```
TypeScript error in /home/giles/ui-example/ui/src/index.tsx(16,8):
No overload matches this call.
  Overload 1 of 2, '(props: Pick<Pick<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "to" | "href" | "media" | "target" | "type" | "className" | "id" | ... 456 more ... | "size"> & RefAttributes<...>, "string" | ... 469 more ... | "size"> & Partial<...>, "string" | ... 469 more ... | "size"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Property 'crossOrigin' is missing in type '{}' but required in type 'Pick<Pick<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "to" | "href" | "media" | "target" | "type" | "className" | "id" | ... 456 more ... | "size"> & RefAttributes<...>, "string" | ... 469 more ... | "size"> & Partial<...>, "string" | ... 469 more ... | "size">'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<ForwardRefExoticComponent<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "to" | "href" | "media" | "target" | "type" | "className" | "id" | "lang" | ... 455 more ... | "size"> & RefAttributes<...>>, any, {}, never>): ReactElement<...>', gave the following error.
    Property 'crossOrigin' is missing in type '{}' but required in type 'Pick<Pick<Pick<StyledIconProps, "string" | "style" | "title" | "clipPath" | "filter" | "mask" | "to" | "href" | "media" | "target" | "type" | "className" | "id" | ... 456 more ... | "size"> & RefAttributes<...>, "string" | ... 469 more ... | "size"> & Partial<...>, "string" | ... 469 more ... | "size">'.  TS2769

    14 |       }}
    15 |     >
  > 16 |       <Up />
       |        ^
    17 |     </div>,
    18 |   document.getElementById('root')
    19 | )

```

ie in one case it requires crossOrigin attribute and in another it requires it not to be there..

**Comment by eugenenelou** [comment]
@jacobwgillespie I am using typescript `3.7.3` and for the `@types` package I assume you mean the react one (according to the README), I use `16.9.15`

I'll try and get a codesandbox working.

**Comment by raulfdm** [comment]
Same here. 

My project is open source, it can be used to see this behaviour:

1. Clone https://github.com/raulfdm/medium-blog
1. Install the dependencies `yarn install`
1. File example: `src/components/InputSearch/InputSearch.tsx` line 44

![image](https://user-images.githubusercontent.com/12464600/74602520-aff53800-50a9-11ea-9b7d-d20b193f4f68.png)

**Comment by jacobwgillespie** [comment]
Okay, I have found the issue - when TypeScript generates the `.d.ts` file for the individual icons, it's getting the generated type wrong and accidentally thinking it should include `crossOrigin`.  I've opened #1152 to fix this, which uses `StyledIcon` as the type - I'll publish this shortly.  Thanks for the reproduction examples!

**Comment by jacobwgillespie** [comment]
This has been published as v9.4.0!

**Comment by raulfdm** [comment]
Thanks @jacobwgillespie . I just upgraded here and now it's working fine. Good job! :)
